### PR TITLE
NSX: Refactor Network & VPC offering

### PR DESCRIPTION
**Pull request by Pearl1594**
### Description

This PR makes necessary changes to the Network offering and VPC offering for NSX

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

File: api/src/main/java/org/apache/cloudstack/api/command/admin/network/CreateNetworkOfferingCmd.java [hunk 1]
```diff
@@ -127,6 +139,18 @@ public class CreateNetworkOfferingCmd extends BaseCmd {
...
 
+    @Parameter(name = ApiConstants.FOR_NSX,
+            type = CommandType.BOOLEAN,
+            description = "true if network offering is meant to be used for NSX, false otherwise.",
+            since = "4.20.0")
+    private Boolean forNsx;
+
+    @Parameter(name = ApiConstants.MODE,
+            type = CommandType.STRING,
+            description = "Indicates the mode with which the network will operate. Valid option: NAT or Route",
```
Comment (nvazquez) [inline api/src/main/java/org/apache/cloudstack/api/command/admin/network/CreateNetworkOfferingCmd.java:150]: Maybe we want to state its only when fornsx=true? We can also do a validation of the parameters (should be ignored if fornsx is not set?)

File: api/src/main/java/org/apache/cloudstack/api/command/admin/network/CreateNetworkOfferingCmd.java [hunk 2]
```diff
@@ -241,6 +282,14 @@ public Boolean getForVpc() {
         return forVpc;
     }
 
+    public Boolean getForNsx() {
```
Comment (nvazquez) [inline api/src/main/java/org/apache/cloudstack/api/command/admin/network/CreateNetworkOfferingCmd.java:285]: Very minor one, but can we name this `isForNsx` and make it return a boolean using the BooleanUtils to make it false when the parameter is null?

File: api/src/main/java/org/apache/cloudstack/api/command/admin/vpc/CreateVPCOfferingCmd.java [hunk 3]
```diff
@@ -120,21 +145,43 @@ public String getDisplayText() {
...
+                    SourceNat.getName(),
+                    NetworkACL.getName(),
+                    PortForwarding.getName(),
+                    UserData.getName()
+                    );
+        }
         return supportedServices;
     }
 
+    public Boolean getForNsx() {
```
Comment (nvazquez) [inline api/src/main/java/org/apache/cloudstack/api/command/admin/vpc/CreateVPCOfferingCmd.java:166]: Same for this

File: engine/schema/src/main/resources/META-INF/db/schema-41810to41900.sql [hunk 4]
```diff
@@ -259,6 +262,7 @@ SELECT
     `network_offerings`.`for_vpc` AS `for_vpc`,
     `network_offerings`.`for_tungsten` AS `for_tungsten`,
     `network_offerings`.`for_nsx` AS `for_nsx`,
+    `network_offerings`.`mode` AS `mode`,
```
Comment (nvazquez) [inline engine/schema/src/main/resources/META-INF/db/schema-41810to41900.sql:265]: I think we can call it nsx_mode as other types will have this field null by default

File: api/src/main/java/org/apache/cloudstack/api/command/admin/network/CreateNetworkOfferingCmd.java (hunk 1 above)
Comment (Pearl1594) [reply api/src/main/java/org/apache/cloudstack/api/command/admin/network/CreateNetworkOfferingCmd.java:150]: I am not sure how to add a validation - based on another parameter in the class..

**Review by nvazquez — APPROVED** [review]
LGTM thanks @Pearl1594

**Pull request merged**